src/shogun/base/SGObject.h

```diff
@@ -394,7 +394,11 @@ class CSGObject
 	template <class T,
 		      class X = typename std::enable_if<is_sg_base<T>::value>::type,
 		      class Z = void>
+#ifdef SWIG
 	void put(const std::string& name, T* value)
+#else
+	void put(std::string_view name, T* value)
```

Comment:
Added string_view for sgobject. There are no typemaps for it yet for swig so left std string signature for swig